packages/manager/src/layouts/OAuth.tsx

```diff
@@ -166,18 +136,29 @@ export class OAuthCallbackPage extends Component<CombinedProps, State> {
     } catch (error) {
       clearStorageAndRedirectToLogout();
     }
-  }
+  };
+
+  React.useEffect(() => {
+    if (!location.search || location.search.length < 2) {
+      clearStorageAndRedirectToLogout();
+      return;
+    }
 
-  render() {
-    const { isLoading } = this.state;
+    const { code, returnTo, state: nonce } = getQueryParamsFromQueryString(
+      location.search
+    );
 
-    if (isLoading) {
-      return <SplashScreen />;
+    if (!code || !returnTo || !nonce) {
+      clearStorageAndRedirectToLogout();
+      return;
     }
 
-    return null;
-  }
-}
+    exchangeAuthorizationCodeForToken(code, returnTo, nonce);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [location.search, dispatchStartSession, history]);
```

Comment:
Any reason not to have an empty dependency array here so it is more inline with `componentDidMount`?  

```suggestion
  }, []);
```

I don't think it really matters in relation to actual functionality, so this is fine